uvm/impl/com/untangle/uvm/SystemManagerImpl.java

```diff
@@ -725,6 +725,66 @@ public boolean upgradesAvailable()
         return upgradesAvailable(true);
     }
 
+    /**
+     * This test all the risks which might cause upgrade failure before actual upgrade starts
+     * 
+     * @return List of risks which might cause upgrade failure before actual upgrade starts 
+     */
+    public List<UpgradeFailures> canUpgrade() 
+    {
+        List<UpgradeFailures> upgradeIssues = new ArrayList<>();
```

Comment:
Can we make it as `Set` instead of `List`, so that it won't take multiple `FAILED_TO_TEST` ?